### DISCUSSION TITLE
Interleaved mapbox and deck layers not working as expected when upgrading to 8.8.x

### DISCUSSION THREAD

**Discussion by ssaornil** [visgl/deck.gl#7170] [Q&A]
We are trying to upgrade deck.gl to the latest version. We are now in 8.7.12 with everything working just fine. We are using deck.gl layers interleaved with Mapbox layers in the same WebGL context. 

We are still using older versions for both mapboxgl and react-map-gl. When we try to upgrade to deck 8.8.x we observe the following behavior:

- Whenever we interact with the viewport (zoom, panning...) the mapbox base map and the layers disappear. Only the deck layers remain until the viewport transition is done.
- The order of the layers is not respected. Deck layers appear always on top.

We created this codesandbox to exemplify what we mean and in the same way we are using it. https://codesandbox.io/s/deck-react-map-interleaved-layers-jyvogx

You'll see we have a `MapCanvas` component that takes care of sorting and differentiating deck layers from mapbox layers and it also takes care of adding/moving the deck layers in mapbox.

Any clue as to what we are doing wrong?

**Comment by Pessimistress** [comment]
The sandbox does not run for me.

From the symptoms that you described, it appears that Deck and Mapbox are not aware of each other. You should check if `map.addLayer(new MapboxLayer(...))` is actually executed.

The mapbox integration module underwent heavy development in 8.8. While the usage pattern in your app is still supported, we are now recommending [MapboxOverlay](https://deck.gl/docs/api-reference/mapbox/mapbox-overlay) with `interleaved: true`. If you migrate to the new pattern, you should not need to implement your own layer sorting.

With that said, it's possible that there has been a regression in the old usage pattern. If you can make the codesandbox work I'm happy to look into it. I tested https://github.com/visgl/deck.gl/blob/8.8-release/test/apps/mapbox-layers/react-app.js and was not able to reproduce your issue.

**Comment by ssaornil** [comment]
@Pessimistress thank you for replying and taking the time to take a look. I fixed the sandbox.

I would love to keep using the same pattern. I saw that new `MapboxOverlay` there but I also read it might be a bit tricky to have multiple map views with that approach and although we are not there right now it is something that we plan on using.

**Reply by chrisgervang** [reply]
With both solutions we can only ever have one interleaved mapbox basemap, since the libraries don't support interleaving multiple mapbox instances onto one deck instance. However, you can have multiple MapViews and show the mapbox map on one of them. This behavior true for MapboxOverlay and MapboxLayer.

**Reply by ssaornil** [reply]
Our current map sandwitch looks like this (from bottom to top)

* Base map
* Deck layer
* mapbox layers
* Deck layer
* mapbox layers

We only use one webGL canvas shared between one instance of DeckGL and one instance of the react-map-gl Map.  

With version 8.7.x this is still working as expected. Do you mean this is not supported anymore?

**Comment by ssaornil** [comment]
This looks like a regression to me so should I report it as a bug?

**Reply by Pessimistress** [reply]
This should be fixed by v8.8.8.
